Build fails with vdr 2.7.1

GetEvent has been deprecated
- ref:
- http://git.tvdr.de/?p=vdr.git;a=commitdiff;h=8d82b050712de0db83ac3b28d04a59479b024622;hp=930e3b4200e0bdf46264ef663e6121264ef6e2e4

```
BUILD      vdr-plugin-epgsearch (target)
    TOOLCHAIN      manual
CC mail.o
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -march=x86-64-v3 -Wall -pipe  -O2 -fomit-frame-pointer -DNDEBUG  -fPIC -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -c -DSENDMAIL='"/usr/sbin/sendmail"' -DHAVE_PCRE2POSIX -DPLUGIN_NAME_I18N='"epgsearch"'  -o mail.o mail.c
CC searchtimer_thread.o
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -march=x86-64-v3 -Wall -pipe  -O2 -fomit-frame-pointer -DNDEBUG  -fPIC -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -c -DSENDMAIL='"/usr/sbin/sendmail"' -DHAVE_PCRE2POSIX -DPLUGIN_NAME_I18N='"epgsearch"'  -o searchtimer_thread.o searchtimer_thread.c
CC switchtimer.o
make: Nothing to be done for 'install-i18n'.
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -march=x86-64-v3 -Wall -pipe  -O2 -fomit-frame-pointer -DNDEBUG  -fPIC -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -c -DSENDMAIL='"/usr/sbin/sendmail"' -DHAVE_PCRE2POSIX -DPLUGIN_NAME_I18N='"epgsearch"'  -o switchtimer.o switchtimer.c
switchtimer.c: In member function 'const cEvent* cSwitchTimer::Event()':
switchtimer.c:130:31: error: 'const class cSchedule' has no member named 'GetEvent'; did you mean 'DelEvent'?
  130 |             event = Schedule->GetEvent(eventID, startTime);
      |                               ^~~~~~~~
      |                               DelEvent
make: *** [Makefile:202: switchtimer.o] Error 1
make: *** Waiting for unfinished jobs....
searchtimer_thread.c: In member function 'void cSearchTimerThread::CheckManualTimers()':
searchtimer_thread.c:921:49: error: 'const class cSchedule' has no member named 'GetEvent'; did you mean 'DelEvent'?
  921 |                 const cEvent* event = schedule->GetEvent(eventID);
      |                                                 ^~~~~~~~
      |                                                 DelEvent
make: *** [Makefile:202: searchtimer_thread.o] Error 1
mail.c: In member function 'virtual const cEvent* cMailTimerNotification::GetEvent() const':
mail.c:78:22: error: 'const class cSchedule' has no member named 'GetEvent'; did you mean 'DelEvent'?
   78 |     return schedule->GetEvent(eventID);
      |                      ^~~~~~~~
      |                      DelEvent
make: *** [Makefile:202: mail.o] Error 1
FAILURE: s/build vdr-plugin-epgsearch during make_target (package.mk)
```

--- a/mail.c
+++ b/mail.c
@@ -75,7 +75,11 @@ const cEvent* cMailTimerNotification::GetEvent() const
     if (!Schedules) return NULL;
     const cSchedule *schedule = Schedules->GetSchedule(channelID);
     if (!schedule) return NULL;
+#if APIVERSNUM >= 20502
+    return schedule->GetEventById(eventID);
+#else
     return schedule->GetEvent(eventID);
+#endif
 }
 
 bool cMailTimerNotification::operator< (const cMailTimerNotification &N) const

--- a/searchtimer_thread.c
+++ b/searchtimer_thread.c
@@ -918,7 +918,11 @@ void cSearchTimerThread::CheckManualTimers(void)
                 if (szEventID)
                     eventID = atol(szEventID);
                 LogFile.Log(3, "checking manual timer %d by event ID %u", ti->Id(), eventID);
+#if APIVERSNUM >= 20502
+                const cEvent* event = schedule->GetEventById(eventID);
+#else
                 const cEvent* event = schedule->GetEvent(eventID);
+#endif
                 if (event) {
                     if (event->StartTime() - bstart != ti->StartTime() || event->EndTime() + bstop != ti->StopTime())
                         ModifyManualTimer(event, ti, bstart, bstop);

--- a/switchtimer.c
+++ b/switchtimer.c
@@ -127,7 +127,11 @@ const cEvent* cSwitchTimer::Event()
         if (!Schedules) return NULL;
         const cSchedule *Schedule = Schedules->GetSchedule(channelID);
         if (Schedule) {
+#if APIVERSNUM >= 20502
+            event = Schedule->GetEventByTime(startTime);
+#else
             event = Schedule->GetEvent(eventID, startTime);
+#endif
             if (!event)
                 event = Schedule->GetEventAround(startTime);
         }

